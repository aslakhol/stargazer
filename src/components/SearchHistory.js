import React from 'react';
import {
  Row,
  Button,
  ListGroup,
  ListGroupItem,
} from 'reactstrap';

import { SEARCH_HISTORY_ENTRIES } from '../utils/constants';

const SearchHistory = (props) => {
  const {
    showHistory,
    isOpen,
    searches,
    clickHistoryItem,
  } = props;

  return (
    <div className="mx-auto">
      <Row>
        <Button color="primary" onClick={() => showHistory()} className="mx-auto">HISTORY</Button>
      </Row>
      <Row>
        <ListGroup className="my-3">
          {isOpen
            && searches.slice(0, SEARCH_HISTORY_ENTRIES).map(search => (
              <ListGroupItem
                key={search.id}
                onClick={() => clickHistoryItem(search.search_string)}
                style={{ minWidth: '95px', textAlign: 'center' }}
              >
                <span>{search.search_string}</span>
              </ListGroupItem>
            ))
          }
        </ListGroup>
      </Row>
    </div>
  );
};

export default SearchHistory;
