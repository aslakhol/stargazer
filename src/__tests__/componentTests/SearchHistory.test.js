import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchHistory from '../../components/SearchHistory';

describe('<SearchHistory />', () => {
  test('renders the component', () => {
    const wrapper = shallow(<SearchHistory />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
