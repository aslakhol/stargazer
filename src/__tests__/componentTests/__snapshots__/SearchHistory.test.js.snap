// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchHistory /> renders the component 1`] = `
<div
  className="mx-auto"
>
  <Row
    tag="div"
  >
    <Button
      className="mx-auto"
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      HISTORY
    </Button>
  </Row>
  <Row
    tag="div"
  >
    <ListGroup
      className="my-3"
      tag="ul"
    />
  </Row>
</div>
`;
